### DISCUSSION TITLE
Should Quadlet [Image] file support Containerfile and Dockerfile?

### DISCUSSION THREAD

**Discussion by khanetor** [containers/podman#22358] [Ideas]
### I will first describe the scenario where I find this to be useful:

Suppose that I have a home data warehouse, where I have Airflow to run all kind of tasks. There is a Docker Image for Airflow already, and to keep things simple, we shall use a LocalExecutor, meaning that the worker is on the same instance as the Airflow web server (a usual Airflow setup will have web server, scheduler, workers, and trigger in separate processes). In order for the worker to actually do works that involves extra libraries, such as Numpy, Pandas, we need to install them. Now, instead of using the original Airflow image, I will need to build a custom image with the dependencies installed. So I have a few options:
- Create a GitHub repository, and have a GitHub Action to build and deploy the custom Airflow image to GitHub Package, which will be consumed by Podman (with autoUpdate).
- Manually build the image locally, and have Podman use the local image instead (also with autoUpdate).

This is manageable, because we only need to rebuild the image every time we have new dependencies. However, if we have Quadlet [Image] to support Containerfile, it should be able to pickup changes in the Containerfile and rebuild the image, saving us the trouble of rebuilding the image ourselves, or pushing to GitHub ourselves.

**Comment by Luap99** [comment]
dup of https://github.com/containers/podman/discussions/17043
